### PR TITLE
Add extra offset when scrolling to a section in Documentati…

### PR DESCRIPTION
**Pull request by dobromir-hristov**
Bug/issue #, if applicable: 100028167

## Summary

Adds an extra offset to the scrollTo helper function, only for the documentation page.

## Dependencies

NA

## Testing

Steps:
1. Click on links in OnThisPage nav or other anchors on the page
2. Assert nav is accounted for + extra 10px for doc pages

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.

- [x] Added tests
- [x] Ran `npm test`, and it succeeded
- [x] Updated documentation if necessary

### CODE REVIEW

File: src/utils/scroll-offset.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+import { documentationTopicName } from 'docc-render/constants/router';
+
+export const EXTRA_DOCUMENTATION_OFFSET = 10;
+
+/**
+ * @param {import('vue-router').Route} to
+ * @returns {Number}
+ */
+export default function getExtraScrollOffset(to) {
```
Comment (mportiz08) [inline src/utils/scroll-offset.js:19]: Would it make sense to move this to the existing `router-utils.js` file where we already have utilities related to scrolling (and also offsets for scrolling) as opposed to creating an entirely new `scroll-offset.js` file?

File: src/utils/scroll-offset.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+ * Copyright (c) 2021 Apple Inc. and the Swift project authors
+ * Licensed under Apache License v2.0 with Runtime Library Exception
+ *
+ * See https://swift.org/LICENSE.txt for license information
+ * See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import { documentationTopicName } from 'docc-render/constants/router';
+
+export const EXTRA_DOCUMENTATION_OFFSET = 10;
```
Comment (mportiz08) [inline src/utils/scroll-offset.js:13]: Can you add a small comment to point out that this is meant to represent pixel units?

File: src/utils/scroll-offset.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+/**
+ * This source file is part of the Swift.org open source project
+ *
+ * Copyright (c) 2021 Apple Inc. and the Swift project authors
```
Comment (mportiz08) [inline src/utils/scroll-offset.js:4]:
```suggestion
 * Copyright (c) 2022 Apple Inc. and the Swift project authors
```

File: src/utils/scroll-offset.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+export const EXTRA_DOCUMENTATION_OFFSET = 10;
+
+/**
+ * @param {import('vue-router').Route} to
+ * @returns {Number}
+ */
+export default function getExtraScrollOffset(to) {
+  const { name } = to;
+  const isDocumentation = name.includes(documentationTopicName);
```
Comment (mportiz08) [inline src/utils/scroll-offset.js:21]: Is it safe to assume that `name` will never be undefined here? I think it is—just double checking.

Comment (dobromir-hristov) [reply src/utils/scroll-offset.js:21]: Yep, we always have a name to our routes.

File: src/utils/scroll-offset.js (hunk 1 above)
Comment (dobromir-hristov) [reply src/utils/scroll-offset.js:23]: Thought it might be easier for future 3rd party users to extend this function 🤔

**Review by mportiz08 — APPROVED** [review]

**Comment by dobromir-hristov** [comment]
@swift-ci test

**Pull request merged**